lib/plugins/aws/package/compile/events/apiGateway/index.js

```diff
@@ -20,15 +21,256 @@ const getMethodAuthorization = require('./lib/method/authorization');
 const getMethodIntegration = require('./lib/method/integration');
 const getMethodResponses = require('./lib/method/responses');
 
+function caseInsensitive(str) {
+  return { type: 'string', regexp: new RegExp(`^${str}$`, 'i').toString() };
+}
+
+const requestParametersSchema = {
+  type: 'object',
+  additionalProperties: {
+    oneOf: [
+      { type: 'boolean' },
+      {
+        type: 'object',
+        properties: {
+          required: { type: 'boolean' },
+          mappedValue: { type: 'string' },
+        },
+        additionalProperties: false,
+      },
+    ],
+  },
+};
+
+const contentHandlingSchema = { enum: ['CONVERT_TO_BINARY', 'CONVERT_TO_TEXT'] };
+
+const allowedMethods = new Set(['GET', 'POST', 'PUT', 'PATCH', 'OPTIONS', 'HEAD', 'DELETE', 'ANY']);
+const methodPattern = new RegExp(`^(?:\\*|${Array.from(allowedMethods).join('|')})$`, 'i');
+const methodPathPattern = new RegExp(
+  `^(?:\\*|(${Array.from(allowedMethods).join('|')}) (\\/\\S*))$`,
+  'i'
+);
+
+const authorizerSchema = {
+  oneOf: [
+    { type: 'string' },
+    caseInsensitive('aws_iam'),
+    { $ref: '#/definitions/awsArn' },
+    {
+      type: 'object',
+      properties: {
+        type: {
+          oneOf: ['token', 'cognito_user_pools', 'request', 'aws_iam'].map(caseInsensitive),
+        },
+        authorizerId: {
+          oneOf: [{ type: 'string' }, { $ref: '#/definitions/awsCfRef' }],
+        },
+        arn: { $ref: '#/definitions/awsArn' },
+        resultTtlInSeconds: { type: 'integer', maximum: 3600 },
```

Comment:
Let's also set `minimum`

Reply:
I think `minimum` should be `0` because:
> If you specify a value greater than 0, API Gateway caches the authorizer responses

Meaning, setting 0 will disable cache. I have no experience with this, but I guess it's a valid option.

Reply:
Sure, my point was, to not welcome negative values, we should set `minimum: 0` I believe